### ISSUE TITLE
Parse SequenceExpression statements in JavaScript

### ISSUE THREAD

**Issue by FredKSchott** [Polymer/polymer-analyzer#379]
Right now we skip over JavaScript `SequenceExpression`s when we scan for Polymer behaviors. This means we miss any expressions that are joined by a comma, like this:

```js
/**
   * @polymerBehavior Polymer.TestBehavior
   */
Polymer.TestBehaviorImpl={properties:{}},/**
   * @polymerBehavior
   */
Polymer.TestBehavior=[Polymer.TestBehaviorImpl,SomeSimpleBehavior];
```

These `SequenceExpression`s are common in minified code, which means analyzing minified JavaScript is pretty broken right now.

See https://github.com/Polymer/polymer-analyzer/commit/2b293f99ba75460be274d2ed6e51d73e0eb79bfe on the `wip-sequence-expression-behaviors` branch for my progress towards fixing this. We'll need to add a `enterSequenceExpression()` method to `BehaviorVisitor` that can properly parse these. I added a basic version of this, but couldn't figure out how to get the different `initBehavior` & `finishBehavior` methods called in the proper order.

Also, we'll need handle how comments relate to `SequenceExpression`, where leading and trailing comments are on the `SequenceExpression` itself and not its first & last expressions.

/cc @rictic

**Comment by rictic** [comment]
Specifically, we miss behaviors defined in the first assignment expression of a sequence expression – not because we don't parse them – but because the comment is attached to the sequence expression node itself, not the assignment expression as expected.

**Comment by rictic** [comment]
One way to address this would be for the behavior scanner to scan first for any ast node annotated with a jsdoc that has `@polymerExpression` on it, then try to analyze the attached node. At least in that case we could emit a Warning if we were unable to understand the resulting structure.

**Comment by FredKSchott** [comment]
Ohhh so we actually were parsing the first assignment expression? that
would explain the double-initBehavior issue I was running into, if I was
trying to overcompensate by calling it manually.

We may still be able to fix this, then, by using enterSequenceExpression to
fix up the comments. Then the other visitors would work as expected. I'll
give that a quick look tonight/tomorrow morning.

**Comment by rictic** [comment]
Yeah, everything is parsed, and everything gets walked in the current system. When there's a parse error you get a `WarningCarryingException`. Scanning does a walk, and it is very thorough, every node is touched even if it's hidden deep inside a subexpression of a subexpression.

**Comment by FredKSchott** [comment]
yup, that was it. Thanks for the clarification, I don't think I would have noticed that otherwise.

Submitting a PR since I already have the work done
